### DISCUSSION TITLE
Tab button in insert mode

### DISCUSSION THREAD

**Discussion by adiosocelote** [helix-editor/helix#8871] [General] [answered]
In the last version of Helix, tab button was changed to go to the next word. Before, my tab button was to add spaces before a word, so from "word" I could do "    word" with just one tab. Now it is impossible to do it anymore, and "indent" command only indents the whole line, not from the point where your cursor is. I have assigned my tab button to "no_op", but this causes tab to not do anything at all. Also, tab works as I want it to in my shell - adds spaces before the cursor location, so this is clearly a Helix issue.

This really makes me want to quit using Helix, honestly. Having to spam space bar multiple times to indent a word is annoying, and I couldn't find a hack to bring back the old behavior.

Version: helix 23.10, Linux, installed from Pacman.

**Comment by 0x61nas** [comment, marked as answer]
this's because the `smart tab` was enabled by default. You can disable it via `:set smart-tab.enable false` or via [`[editor.smart-tab]`](https://docs.helix-editor.com/configuration.html#editorsmart-tab-section) section in the config file.

**Comment by tmacychen** [comment]
With the default bindings, to explicitly insert a tab character, press Shift-tab. from:  https://docs.helix-editor.com/editor.html#editorsmart-tab-section
